sns.tf

```diff
@@ -38,11 +40,34 @@ resource "aws_sns_topic" "apiary_data_events" {
         "Effect": "Allow",
         "Principal": {"AWS":"*"},
         "Action": "SNS:Publish",
-        "Resource": "arn:aws:sns:*:*:${local.instance_alias}-${local.apiary_managed_schema_names_replaced[count.index]}-data-events",
+        "Resource": "arn:aws:sns:*:*:${local.instance_alias}-${each.value["resource_suffix"]}-data-events",
         "Condition":{
-            "ArnLike":{"aws:SourceArn":"${aws_s3_bucket.apiary_data_bucket.*.arn[count.index]}"}
+            "ArnLike":{"aws:SourceArn":"${aws_s3_bucket.apiary_data_bucket[each.key].arn}"}
         }
     }]
 }
 POLICY
 }
+
+resource "aws_sqs_queue" "apiary_data_event_queue" {
+  count = local.create_sqs_data_event_queue ? 1 : 0
+  name = "${local.instance_alias}-data-event-queue"
+
+  policy = <<POLICY
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {"AWS":"*"},
```

Comment:
is that * ok here ?

Reply:
I was just following the IAM policy for the SNS topic: https://github.com/ExpediaGroup/apiary-data-lake/blob/master/sns.tf#L39

However, you are right, I should probably tighten it up as per this doc: https://docs.aws.amazon.com/AmazonS3/latest/dev/NotificationHowTo.html#grant-destinations-permissions-to-s3.  Will do.

Reply:
Done